### DISCUSSION TITLE
Is the motion zone name exposed at all?

### DISCUSSION THREAD

**Discussion by jcallaghan** [briis/unifiprotect#327] [Q&A]
I leverage the smart detection zones for person and vehicle detection but I also have G3 cameras which don't have smart detection zones. On these cameras, I have mapped out different motion zones. When a motion zone is tripped, is the name of the zone that was tripped exposed at all? 

Here is an example, I have setup three zones for my pets food and water bowls. If any one of these zones is tripped then the motion binary_sensor becomes active but if the name of the motion zone is exposed I could have just two motion zones and make some more powerful automation. It would be amazing to have this available as an attribute if the motion zone name is exposed! How could I help answer my own question?

![image](https://user-images.githubusercontent.com/7449908/140655762-ba3ca8e5-dad2-4014-ae43-444f56f77928.png)

**Comment by AngellusMortis** [comment]
That data is not currently exposed, but it is in theory possible to get it. None of the data is really current exposed in HA so you would definitely need to make a feature request for us to add it.

Details for the feature request (or if you just want to poke around):

You can get the smart detect data for an event at `/proxy/protect/api/events/{event_id}/smartDetectTrack`. These has the data the UFP uses to draw the boxes around the detections in the UFP app. The `zones` arg here is what is of interest for what you want. That gives the `id` that maps back to the `smartDetectZones` on the camera itself.

**Reply by jcallaghan** [reply]
Amazing, thank you. I'd certainly like to poke around some more and perhaps contribute to making this a possibility. Is there a log or a way to track down the ```event_id``` so I can explore this some more, please?

**Reply by AngellusMortis** [reply]
Looking more at it, it actually looks like the specific data you want is _not_ exposed. What I was talking about was specifically for Smart Detections, not Motion events. They are very similar in concept, but they are actually both implemented very different in UFP. 

As far as I can tell, UFP does not expose which motion zone triggered a motion event like it does for a smart detection. You only get a heatmap of the motion activity. 

Example of what it looks like: 
![image](https://user-images.githubusercontent.com/490848/141845072-c6048c09-8ba7-44ba-9143-87fcb032b634.png)

**Reply by toSimmi** [reply]
> That data is not currently exposed, but it is in theory possible to get it. None of the data is really current exposed in HA so you would definitely need to make a feature request for us to add it.
> 
> Details for the feature request (or if you just want to poke around):
> 
> You can get the smart detect data for an event at `/proxy/protect/api/events/{event_id}/smartDetectTrack`. These has the data the UFP uses to draw the boxes around the detections in the UFP app. The `zones` arg here is what is of interest for what you want. That gives the `id` that maps back to the `smartDetectZones` on the camera itself.

Hi, Did you find a solution to get the information from which detection Zone (person detection) the trigger was comming. i was not able to understand your comment about  `/proxy/protect/api/events/{event_id}/smartDetectTrack`.

**Reply by AngellusMortis** [reply]
It is added to the underlaying Python lib already, but not exposed in HA: https://github.com/briis/pyunifiprotect/blob/master/pyunifiprotect/data/nvr.py#L265

Since the original ask for motion zones, not smart detect zones, I never added it after I figured out how it worked.

**Reply by mullermn** [reply]
I would be really happy to be able to be able to extract which smart detection zone triggered a motion event - and if I follow the discussion above correctly it sounds like this is feasible?

As an example use case, with an external camera and multiple motion zones I could make more complex behaviours based on whether someone is approaching or leaving a building.
 
I have created an issue to track this feature request here: https://github.com/briis/unifiprotect/issues/458
